docs/src/user-guide/tutorials/auction.md

```diff
@@ -0,0 +1,351 @@
+# Auction contract
+
+This tutorial aims to implement a simple auction contract in Fe. Along the way you will learn some foundational Fe concepts.
+
+An open auction is one where prices are determined in real-time by live bidding. The winner is the participant who has made the highest bid at the time the auction ends.
+
+## The auction rules
+
+To run an open auction, you need an item for sale, a seller, a pool of buyers and a deadline after which no more bids will be recognized. In this tutorial we will not have an item per se, the buyers are simply bidding to win! The highest bidder is provably crowned the winner, and the value of their bid is passed to the beneficiary. Bidders can also withdraw their bids at any time.
+
+
+## Get Started
+
+To follow this guide you should have Fe installed on your computer. If you haven't installed Fe yet, follow the instructions on the [Installation](../installation.md) page.
+
+With Fe installed, you can create a project folder, `auction` that will act as your project root. In that folder, create an empty file called `auction.fe`.
+
+Now you are ready to start coding in Fe!
+
+You will also need [Foundry](https://getfoundry.sh) installed to follow the deployment instructions in this guide - you can use your alternative tooling for this if you prefer.
+
+## Writing the Contract
+
+You can see the entire contract [here](../example_contracts/auction_contract.md). You can refer back to this at any time to check implementation details.
+
+### Defining the `Contract` and initializing variables 
+
+A contract is Fe is defined using the `contract` keyword. A contract requires a constructor function to initialize any state variables used by the contract. If no constructor is defined, Fe will add a default with no state variables. The skeleton of the contract can look as follows:
+
+```rust
+contract Auction {
+    pub fn __init__() {}
+}
+```
+
+To run the auction you will need several state variables, some of which can be initialized at the time the contract is instantiated.
+You will need to track the address of the beneficiary so you know who to pay out to. You will also need to keep track of the highest bidder, and the amount they have bid. You will also need to keep track of how much each specific address has sent into the contract, so you can refund them the right amount if they decide to withdraw. You will also need a flag that tracks whether or not the auction has ended. The following list of variables will suffice:
+
+```sh
+auction_end_time: u256
+beneficiary: address
+highest_bidder: address
+highest_bid: u256
+pending_returns: Map<address, u256>
+ended: bool
+```
+
+Notice that variables are named using camel case (lower case, underscore separated, `like_this`). 
```

Comment:
I think the correct term is *snake case* because camel case would be `likeThis`

Reply:
doh - yes of course. Sorry!